sequence_processing_pipeline/SeqCountsJob.py

```diff
@@ -133,15 +145,100 @@ def extract_metadata(fp):
                     _dir, _file, seq_counts, base_pairs = \
                         extract_metadata(log_output_file)
 
-                    if _dir not in results:
-                        results[_dir] = {}
+                    results[_file] = {
+                        'seq_counts': seq_counts,
+                        'base_pairs': base_pairs
+                    }
+
+        return results
+
+    def _aggregate_counts(self, sample_sheet_path):
+        """
+        Aggregate results by sample_ids and write to file.
+        Args:
+            sample_sheet_path:
+
+        Returns: None
+        """
+        def get_metadata(sample_sheet_path):
+            sheet = load_sample_sheet(sample_sheet_path)
+
+            lanes = []
+
+            if sheet.validate_and_scrub_sample_sheet():
+                results = {}
+
+                for sample in sheet.samples:
+                    barcode_id = sample['barcode_id']
+                    sample_id = sample['Sample_ID']
+                    lanes.append(sample['Lane'])
+                    results[barcode_id] = sample_id
+
+                lanes = list(set(lanes))
+
+                if len(lanes) != 1:
+                    raise ValueError(
+                        "More than one lane is declared in sample-sheet")
+
+                return results, lanes[0]
+
+        mapping, lane = get_metadata(sample_sheet_path)
+
+        # aggregate results by filename
+        by_files = self._aggregate_counts_by_file()
+
+        # aggregate results by sample. This means aggregating metadata
+        # across two file names.
+        counts = defaultdict(dict)
+
+        for _file in by_files:
+            if 'erroneous' in _file:
```

Comment:
Should this raise an error?

Reply:
... if not, what about a warning?

Reply:
So the background on this is that we're using the raw fastq files stored in the output directory's 'Full' directory. For the run that we're using as our test, we're generating files with two filename types:    "TellReadJob_I1_C539.fastq.gz.erroneous.fastq"
"TellReadJob_I1_C539.fastq.gz.corrected.err_barcode_removed.fastq"

The 'erroneous' files were not insignificant in size but less than those in the 'corrected' files. The included comment marks it as a tentative decision because it's easily changed to include both or raise an error. @wasade would you please comment?

Reply:
I discussed this with Omar and @wasade in Slack and the recommended course of action was to generate raw_count_r1r2 from the integrated output, rather than immediately after the creation of the raw fastqs. The erroneous files do not appear in the integrated output and this error check was removed.